RecoLocalCalo/EcalRecProducers/plugins/alpaka/EcalUncalibRecHitPhase2WeightsAlgoPortable.dev.cc

```diff
@@ -0,0 +1,111 @@
+#include "DataFormats/EcalDigi/interface/EcalDataFrame_Ph2.h"
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+
+#include "EcalUncalibRecHitPhase2WeightsAlgoPortable.h"
+
+#include "DataFormats/EcalDigi/interface/alpaka/EcalDigiPhase2DeviceCollection.h"
+#include "DataFormats/EcalRecHit/interface/alpaka/EcalUncalibratedRecHitDeviceCollection.h"
+#include <alpaka/alpaka.hpp>
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/traits.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+
+#include "DataFormats/EcalRecHit/interface/EcalUncalibratedRecHit.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  namespace ecal {
+    namespace weights {
+
+      using namespace cms::alpakatools;
+
+      class Phase2WeightsKernel {
+      public:
+        template <typename TAcc>
+        ALPAKA_FN_ACC void operator()(TAcc const &acc,
+                                      double const *weightsdata,
+                                      double const *timeWeightsdata,
+                                      EcalDigiPhase2DeviceCollection::ConstView digisDev,
+                                      EcalUncalibratedRecHitDeviceCollection::View uncalibratedRecHitsDev) const;
+      };
+
+      template <typename TAcc>
+      ALPAKA_FN_ACC void Phase2WeightsKernel::operator()(
+          TAcc const &acc,
+          double const *weightsData,
+          double const *timeWeightsdata,
+          EcalDigiPhase2DeviceCollection::ConstView digisDev,
+          EcalUncalibratedRecHitDeviceCollection::View uncalibratedRecHitsDev) const {
+        constexpr int nsamples = EcalDataFrame_Ph2::MAXSAMPLES;
+        auto const nchannels = digisDev.size();
+        // one thread sets the output collection size scalar
+        if (alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u] == 0) {
+          uncalibratedRecHitsDev.size() = digisDev.size();
+        }
+
+        auto *amplitude = uncalibratedRecHitsDev.amplitude();
+        auto *jitter = uncalibratedRecHitsDev.jitter();
+        const auto *digis = digisDev.data();
+        //calculate the first and the stride
+        const auto first = alpaka::getIdx<alpaka::Block, alpaka::Threads>(acc)[0u] +
+                           alpaka::getIdx<alpaka::Grid, alpaka::Blocks>(acc)[0u] *
+                               alpaka::getWorkDiv<alpaka::Block, alpaka::Threads>(acc)[0u];
+        const auto stride = alpaka::getWorkDiv<alpaka::Block, alpaka::Threads>(acc)[0u] *
+                            alpaka::getWorkDiv<alpaka::Grid, alpaka::Blocks>(acc)[0u];
+
+        for (auto tx = first; tx < nchannels; tx += stride) {
+          bool g1 = false;
+          auto const did = DetId{digisDev.id()[tx]};
+          amplitude[tx] = 0;
+          jitter[tx] = 0;
```

Comment:
You could do also
``cpp
auto hit = uncalibratedRecHitsDev[tx];
hit.amplitude() = 0;
hit.jitter() = 0;
```
etc, and that should lead to same (or very similar) code being generated by the compiler.